sui/src/bin/bench_configure.rs

```diff
@@ -0,0 +1,182 @@
+        ObjectConfigRange,
+    },
+};
+
+use sui_types::{base_types::ObjectID, crypto::get_key_pair};
+
+const BUFFER_SIZE: usize = 650000;
+
+#[derive(Debug, Parser)]
+#[clap(
+    name = "Sui Distributed Benchmark Config Creator",
+    about = "Creates the config files for distributed benchmarking"
+)]
+pub struct DistributedBenchmarkConfigurator {
+    /// List of space separated strings of the form: host:port:stake, example 127.0.0.1:8080:5
+    #[clap(long, multiple_values = true)]
+    pub host_port_stake_triplets: Vec<String>,
+    #[clap(long, default_value = "0x0000000000000000000000000010000000000000")]
+    pub object_id_offset: ObjectID,
+    #[clap(long, default_value = "4")]
+    pub number_of_generators: usize,
+    #[clap(long, default_value = "1000000")]
+    pub number_of_txes_per_generator: usize,
+}
+fn main() {
+    let bch = DistributedBenchmarkConfigurator::parse();
+
+    let mut authorities = vec![];
+    let mut authority_keys = BTreeMap::new();
+
+    // Create configs for authorities
+    for b in bch.host_port_stake_triplets {
+        let (host, port, stake) = parse_host_port_stake_triplet(&b);
+        let (validator_address, validator_keypair) = get_key_pair();
+        let db_path = format!("DB_{}", validator_address);
+        let path = Path::new(&db_path);
+
+        let host_bytes: Vec<u8> = host
+            .split('.')
+            .into_iter()
+            .map(|q| q.parse::<u8>().unwrap())
+            .collect();
+        let consensus_address = SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::new(
+                host_bytes[0],
+                host_bytes[1],
+                host_bytes[2],
+                host_bytes[3],
+            )),
+            port + 1,
+        );
+
+        let auth = AuthorityPrivateInfo {
+            address: validator_address,
+            host,
+            port,
+            db_path: path.to_path_buf(),
+            stake,
+            consensus_address,
+            public_key: *validator_keypair.public_key_bytes(),
+        };
+
+        authorities.push(auth);
+        authority_keys.insert(*validator_keypair.public_key_bytes(), validator_keypair);
+    }
+
+    // For each load generator, create an address as the source of transfers, and configs
+    let mut accounts = vec![];
+    let mut obj_id_offset = bch.object_id_offset;
+    let mut account_private_info = vec![];
+
+    // For each load gen, create an account
+    for _ in 0..bch.number_of_generators {
+        // Create a keypair for this account
+        let (account_address, account_keypair) = get_key_pair();
+
+        // Populate the range configs
+        let range_cfg = ObjectConfigRange {
+            offset: obj_id_offset,
+            count: bch.number_of_txes_per_generator as u64,
+            gas_value: u64::MAX,
+        };
+
+        account_private_info.push((account_keypair, obj_id_offset));
+
+        // Ensure no overlap
+        obj_id_offset = obj_id_offset
```

Comment:
Here again, can we avoid doing arithmetic on object IDs and just generate them at random? I assume this config happens centrally?

Reply:
If we generate a bunch of random object ids, we end up with a genesis config file millions of lines long which then has to be shipped around and parsed by each validator. It slows down startup time.
My solution keeps this simply by using clearly defined ranges